src/libs/sfdk/virtualmachine.h

```diff
@@ -104,15 +104,18 @@ class SFDK_EXPORT VirtualMachine : public QObject
     void setMemorySizeMb(int memorySizeMb, const QObject *context,
             const Functor<bool> &functor);
     static int availableMemorySizeMb();
+    virtual bool memorySizeEditable() const { return true; }
 
     int cpuCount() const;
     void setCpuCount(int cpuCount, const QObject *context,
             const Functor<bool> &functor);
     static int availableCpuCount();
+    virtual bool cpuCountEditable() const { return true; }
 
     int storageSizeMb() const;
     void setStorageSizeMb(int storageSizeMb, const QObject *context,
             const Functor<bool> &functor);
+    virtual bool storageSizeEditable() const { return true; }
```

Comment:
I would prefer adding flags instead

```
enum Feature {
    NoFeatures = 0x00,
    LimitMemorySize = 0x01,
    LimitCpuCount = 0x02,
    GrowStorageSize = 0x04,
    ShrinkStorageSize = 0x08,
    OptionalHeadless = 0x10,
    Snapshots = 0x11,
};
Q_DECLARE_FLAGS(Features, Feature)
...
QString displayType() const;
Features features() const;
QString name() const;
...
    VirtualMachine(std::unique_ptr<VirtualMachinePrivate> &&dd, const QString &type,
            const QString &name, const Features &features, QObject *parent);
```

and QTC_ASSERT their availability in this class already, e.g.:

```
void VirtualMachine::setMemorySizeMb(int memorySizeMb, const QObject *context,
        const Functor<bool> &functor)
{
    Q_D(VirtualMachine);
    QTC_ASSERT(d->features & LimitMemorySize,
            QTimer::singleShot(0, context, std::bind(functor, false)); return);
    ...
}
```